<robot name="robot">
  <link name="map"></link>
  <joint name="base_joint" type="fixed">
    <parent link="map" />
    <child link="base_link" />
    <origin xyz="0 0 4" rpy="0 0 0" />
  </joint>
  <link name="base_link">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <box size="1 1 1" />
      </geometry>
      <material name="base_link-material">
        <color rgba="0.6038273388475408 0.03820437158923601 0.6653872982754769 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <box size="1 1 1" />
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="1" />
      <inertia ixx="0.16666666666666666" ixy="0" ixz="0" iyy="0.16666666666666666" iyz="0" izz="0.16666666666666666" />
    </inertial>
  </link>
  <joint name="base_link_to_uav_bearing" type="continuous">
    <parent link="base_link" />
    <child link="uav_bearing" />
    <origin xyz="0 0 -1" rpy="0 0 0" />
    <axis xyz="0 0 1"/>
  </joint>
  <link name="uav_bearing">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <sphere radius="0.5" />
      </geometry>
      <material name="uav_bearing-material">
        <color rgba="0.5906188409113381 0.0865004620280852 0.08228270712149792 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <sphere radius="0.5" />
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="1" />
      <inertia ixx="0.4" ixy="0" ixz="0" iyy="0.4" iyz="0" izz="0.4" />
    </inertial>
  </link>
  <joint name="uav_bearing_to_uav_banking" type="continuous">
    <parent link="uav_bearing" />
    <child link="uav_banking" />
    <origin xyz="0 0 0" rpy="0 1.5707963267948966 0" />
    <axis xyz="0 0 1"/>
  </joint>
  <link name="uav_banking">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <sphere radius="0.5" />
      </geometry>
      <material name="uav_banking-material">
        <color rgba="0.5457244613615395 0.12743768042608497 0.46778379610254284 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <sphere radius="0.5" />
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="1" />
      <inertia ixx="0.4" ixy="0" ixz="0" iyy="0.4" iyz="0" izz="0.4" />
    </inertial>
  </link>
  <joint name="uav_banking_to_uav_pitching" type="continuous">
    <parent link="uav_banking" />
    <child link="uav_pitching" />
    <origin xyz="0 0 0" rpy="1.5707963267948963 0 0" />
    <axis xyz="0 0 1"/>
  </joint>
  <link name="uav_pitching">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <sphere radius="0.5" />
      </geometry>
      <material name="uav_pitching-material">
        <color rgba="0.2917706498067336 0.2541520943200296 0.6795424696265424 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <sphere radius="0.5" />
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="1" />
      <inertia ixx="0.4" ixy="0" ixz="0" iyy="0.4" iyz="0" izz="0.4" />
    </inertial>
  </link>
  <joint name="uav_pitching_to_cylinder0" type="revolute">
    <parent link="uav_pitching" />
    <child link="cylinder0" />
    <origin xyz="1 0 0" rpy="-1.5707963267948966 0 1.5707963267948963" />
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>  </joint>
  <link name="cylinder0">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.5" length="1" />
      </geometry>
      <material name="cylinder0-material">
        <color rgba="0.057805430183792694 0.0033465357620980074 0.9473065367320066 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.5" length="1" />
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="1" />
      <inertia ixx="0.3333333333333333" ixy="0" ixz="0" iyy="0.5" iyz="0" izz="0.3333333333333333" />
    </inertial>
  </link>
  <joint name="cylinder0_to_cylinder1" type="revolute">
    <parent link="cylinder0" />
    <child link="cylinder1" />
    <origin xyz="1 0 0" rpy="1.5707963267948963 0 5.473822126268817e-48" />
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-1.57" upper="1.57" velocity="0.5"/>  </joint>
  <link name="cylinder1">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.5" length="1" />
      </geometry>
      <material name="cylinder1-material">
        <color rgba="0.48514994004665124 0.21586050010324415 0.0027317428517647058 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.5" length="1" />
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="1" />
      <inertia ixx="0.3333333333333333" ixy="0" ixz="0" iyy="0.5" iyz="0" izz="0.3333333333333333" />
    </inertial>
  </link>
  <joint name="cylinder1_to_cylinder2" type="continuous">
    <parent link="cylinder1" />
    <child link="cylinder2" />
    <origin xyz="1 0 0" rpy="-6.077163357286271e-64 1.5707963267948966 -6.077163357286271e-64" />
    <axis xyz="0 0 1"/>
  </joint>
  <link name="cylinder2">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.5" length="1" />
      </geometry>
      <material name="cylinder2-material">
        <color rgba="0.6307571363387763 0.2704977910022518 0.4910208498384856 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.5" length="1" />
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="1" />
      <inertia ixx="0.3333333333333333" ixy="0" ixz="0" iyy="0.5" iyz="0" izz="0.3333333333333333" />
    </inertial>
  </link>
  <joint name="cylinder2_to_cube1" type="fixed">
    <parent link="cylinder2" />
    <child link="cube1" />
    <origin xyz="0 0 1" rpy="-1.5707963267948966 -1.5707963267948966 0" />
  </joint>
  <link name="cube1">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <box size="1 1 1" />
      </geometry>
      <material name="cube1-material">
        <color rgba="0.20863687013464577 0.5394794890033748 0.19461783043107173 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <box size="1 1 1" />
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="1" />
      <inertia ixx="0.16666666666666666" ixy="0" ixz="0" iyy="0.16666666666666666" iyz="0" izz="0.16666666666666666" />
    </inertial>
  </link>
</robot>